src/keys/bip39.rs

```diff
@@ -43,17 +165,43 @@ pub mod wordlist {
     #[cfg_attr(docsrs, doc(cfg(feature = "bip39-jp")))]
     include!("bip39.jp.rs");
 
-    #[derive(Debug, PartialEq, Eq)]
-    pub enum Error {
-        InvalidEntropyCount(usize),
-        NoSuchWord(String),
-        ChecksumMismatch,
-    }
-
     const fn cs(ent: usize) -> usize {
         ent / 32
     }
 
+    impl<'a> Wordlist<'a> {
+        // TODO: should it be pub?
+        const fn new_unchecked(separator: char, words: &'a [&'a str; 2048]) -> Self {
+            Self { words, separator }
+        }
+
+        pub fn new(separator: char, words: &'a [&'a str; 2048]) -> Result<Self, Error> {
```

Comment:
Just curious, what's the usecase of actually taking a separator as parameter? Can't we just use one internally?

Reply:
Don't know. Historical reasons probably. Japanese wordlist uses U+3000 as space but it's still normalizes to U+0020. Separator is a tab or a new-line char. Different separators result in different derived seeds. So it makes sense to have is as parameter.